src/lib/transition_frontier_controller_tests/test_transition_frontier_persistence.ml

```diff
@@ -50,10 +50,14 @@ let%test_module "Transition Frontier Persistence" =
       Monitor.try_with (fun () ->
           let frontier_persistence = create_persistence ~directory_name in
           let%bind result = f (frontier, frontier_persistence) in
-          let%map () =
-            Transition_frontier_persistence.close_and_finish_copy
-              frontier_persistence
+          frontier_persistence.is_accepting_diffs <- false ;
+          let%bind () =
+            Transition_frontier_persistence.flush frontier_persistence
           in
+          Strict_pipe.Writer.close frontier_persistence.worker_writer ;
```

Comment:
Once again, the details of how to shutdown the frontier persistence should be encapsulated internally to the module that defines it's behavior. It would be better to put all this logic in the persistence module and have a `stop` hook or something.